lib/jsdom/level2/html.js

```diff
@@ -2096,6 +2097,36 @@ define('HTMLFrameElement', {
         }
       }
     },
+
+    _detach: function () {
+      core.HTMLElement.prototype._detach.call(this);
+
+      var self = this;
+      var parent = this._ownerDocument.defaultView;
+
+      if (self.contentWindow) {
+        self.contentWindow.close();
+      }
+
+      // delete frame reference
+      if (self.getAttribute("name")) {
+        parent._frame(self.getAttribute("name"));
+      }
+
+      // delete index reference by moving all frames with a higher index one index down
+      for (var i = self._frameID; i < parent.length - 1; ++i) {
+        var next = parent.document.querySelectorAll("iframe,frame")[i];
```

Comment:
Moved = detach + attach, right? I think we should have enough hooks to make this work...

This does seem sketchy though, switching between the existing `parent` pseudo-array and `parent.document.querySelectorAll("iframe,frame")`; it's not clear which is the source of truth. Maybe just renumber everything from scratch (based on QSA results) whenever any frame is attached/detached? Probably simpler, and rare enough that perf shouldn't matter.